src/types/UiaTextRangeBase.cpp

```diff
@@ -990,1534 +902,359 @@ const COORD UiaTextRangeBase::_getScreenFontSize() const
 }
 
 // Routine Description:
-// - Gets the number of rows in the output text buffer.
+// - Gets the viewport height, measured in char rows.
 // Arguments:
-// - pData - the UiaData for the terminal we are operating on
+// - viewport - The viewport to measure
 // Return Value:
-// - The number of rows
-const unsigned int UiaTextRangeBase::_getTotalRows(gsl::not_null<IUiaData*> pData) noexcept
+// - The viewport height
+const unsigned int UiaTextRangeBase::_getViewportHeight(const SMALL_RECT viewport) noexcept
 {
-    return pData->GetTextBuffer().TotalRowCount();
+    FAIL_FAST_IF(!(viewport.Bottom >= viewport.Top));
+    // + 1 because COORD is inclusive on both sides so subtracting top
+    // and bottom gets rid of 1 more then it should.
+    return viewport.Bottom - viewport.Top + 1;

```

Comment:
If `Bottom` and `Top` are both `SHORT`, then can't this technically cast into something gross when it's returned as `unsigned int`?

Reply:
This can technically have a range of [1, SHORT_MAX+1] because of that FailFast assertion. So, I think this is correct as-is? We need unsigned int for that extra 1...